src/SourceLocator/Type/ComposerSourceLocator.php

```diff
@@ -22,9 +24,9 @@ class ComposerSourceLocator extends AbstractSourceLocator
      */
     private $classLoader;
 
-    public function __construct(ClassLoader $classLoader)
+    public function __construct(ClassLoader $classLoader, ?Locator $locator = null)
```

Comment:
IMO, the dependency should be the parser, and the dependency shouldn't be optional.

Reply:
Don't make the dep optional then ;)